### ISSUE TITLE
Is it possible to configure the MsalModule not with the forRoot()-Method?

### ISSUE THREAD

**Issue by SebastianFreund** [Marcelh1983/angular-msal#1]
Hi, 
we want to configure the MsalModule with values from a config-file.
But we haven't found a way to get the values in into the object, which is used in the forRoot() method.

Maybe it is possible to configure it like this:

```
providers: [
            { 
                provide: MSAL_CONFIG, 
                useFactory: getConfig
            }, 
```
```
export function getConfig(): MsalConfig {
    return { ... }
}
```

In the getConfig function we can access the config.

But to make this working, you have to make the MSAL_CONFIG accessible. Is this possible?
Or maybe you know another way to configure the MsalModule with values from a config-file?

**Comment by Marcelh1983** [comment]
Hi Sebastian,
I can have a look at this but I do prefer not to do as little changes from the original package as possible.
environment.ts is use as a config file,. Why can't you configure the msal configuration options there?

**Comment by SebastianFreund** [comment]
Hi Marcel,

the application is used by different customers. So it should be possible to modify the config without changing the sourcecode  --> config should be in a json file.
And unfortunately the config couldn't be loaded before the module is initalized...

**Comment by Marcelh1983** [comment]
Hi Sebastian,
I get what you mean and it id definitely useful. It could be like how [ngxTranslate   ](https://github.com/ngx-translate/core/blob/2c2679329643e2fb4eb7c6e8515b536387f6fefb/projects/ngx-translate/core/src/public_api.ts) get it's translation file. I started a branch [config-loader](https://github.com/Marcelh1983/angular-msal/tree/config-loader) but it might be a problem that msalService extended UserAgentApplication and expected the config in the constructor. I don't know what happens if you retrieve it async. I started this branch but it will cost me too much time to continue now. Feel free to clone it and try to finish it. Otherwise i'll see if I have time to continue..
